mobile/test/common/extensions/cert_validator/platform_bridge/platform_bridge_cert_validator_test.cc

```diff
@@ -224,6 +224,38 @@ TEST_P(PlatformBridgeCertValidatorTest, ValidCertificate) {
   EXPECT_FALSE(waitForDispatcherToExit());
 }
 
+TEST_P(PlatformBridgeCertValidatorTest, ValidCertificateEmptySocketOptions) {
+  initializeConfig();
+  PlatformBridgeCertValidator validator(&config_, stats_, &platform_validator_);
+
+  std::string hostname = "server1.example.com";
```

Comment:
Ah, great question! So `auto` would definitely work and it would make the code smaller. But the style guide says:

> The fundamental rule is: use type deduction only to make the code clearer or safer, and do not use it merely to avoid the inconvenience of writing an explicit type

https://google.github.io/styleguide/cppguide.html#Type_deduction

Basically I tend to avoid using `auto` unless the type is an iterator (in which case the type is somewhat unreadable anyway, and often redundant) or if i'm doing something like `std::make_unique<SomeType>()` in which case the type is totally obvious. Otherwise, I tend to just type out the type. In particular, I sometimes find `auto` used to avoid typing out the name of some proto object. Those type names are long, but when I'm trying to understand some code I often want to understand "What fields does this variable have" and `auto` doesn't let me do the kind of code searching I'd ideally like to do.

Anyway, that's my philosophy :) Other folks have different takes...

Reply:
I'm 100% aligned with Ryan's philosophy here, FWIW! Also the style guide says so :)